bootstrap/git_test.go

```diff
@@ -1,54 +1,72 @@
 package bootstrap
 
 import (
-	"os"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
-func TestParsingGittableRepository(t *testing.T) {
+func TestParsingGittableRepositoryFromFilesPaths(t *testing.T) {
 	t.Parallel()
 
-	wd, _ := os.Getwd()
-
-	var testCases = []struct {
-		Ref      string
-		Expected string
-		IsErr    bool
-	}{
-		{wd, wd, false},
-		{"git@github.com:buildkite/agent.git", "ssh://git@github.com/buildkite/agent.git", false},
-		{"git@github.com-alias1:buildkite/agent.git", "ssh://git@github.com-alias1/buildkite/agent.git", false},
-		{"ssh://git@scm.xxx:7999/yyy/zzz.git", "ssh://git@scm.xxx:7999/yyy/zzz.git", false},
-		{"ssh://root@git.host.de:4019/var/cache/git/project.git", "ssh://root@git.host.de:4019/var/cache/git/project.git", false},
+	u, err := parseGittableURL(`/home/vagrant/repo`)
+	if err != nil {
```

Comment:
Should this be...

```go
 assert.Nil(err)
```

..instead?

Reply:
I also learnt about this syntax:

```
func TestSomething(t *testing.T) {
  assert := assert.New(t)

  assert.Nil(err)
}
```

So you don't need to pass `t` around everywhere.

Reply:
The problem with that is that assert.Nil isn't actually fatal, it doesn't stop the test.  I tend to think the if block with a Fail is a lot clearer here. Go's error handling is meant to look like this :)

Reply:
The alternative would be:

```
if !assert.NoError(err) {
  t.FailNow()
}
```

Which is ok I guess? Doesn't seem worth the cognitive overhead.

Reply:
Orrr....

```go
import (
  "github.com/stretchr/testify/require"
)

func TestSomething(t *testing.T) {
  require := require.New(t)

  require.Nil(err)
}
```

That'll blow up the test (instead of running) if it fails! Maybe better/worse?

Reply:
(See: https://godoc.org/github.com/stretchr/testify/require)

Reply:
It makes me sad :(

Reply:
Reckon I'm going to stick with this as is rather than bring in a whole new package.

Reply:
Can you just use `require` instead of `assert` for the whole thing?

Reply:
Yup, we could. I guess if we are doing the whole testify thing we might as well bring it all in.